cdap-security-spi/src/main/java/io/cdap/cdap/security/spi/authorization/AuthorizationResult.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.spi.authorization;
+
+import java.util.Set;
+import javax.annotation.Nullable;
+
+public class AuthorizationResult {
+
+  public enum AuthorizationStatus {
+    AUTHORIZED,
+    UNAUTHORIZED,
+    NOT_REQUIRED
+  }
+
+  private final AuditLogContext auditLogContext;
+  private final AuthorizationStatus authorized;
+  // Fields to handle Unauthorized exception
+
+  private final String principal;
+  private final Set<String> missingPermissions;
+  private final String entity;
+  private final Throwable ex;
+  private final boolean requiresAllPermissions;
+  private final boolean includePrincipal;
+  private final String addendum;
+
+  private AuthorizationResult(Builder builder) {
+    this.auditLogContext = builder.auditLogContext;
+    this.authorized = builder.authorized;
+    this.principal = builder.principal;
+    this.missingPermissions = builder.missingPermissions;
+    this.entity = builder.entity;
+    this.ex = builder.ex;
+    this.requiresAllPermissions = builder.requiresAllPermissions;
+    this.includePrincipal = builder.includePrincipal;
+    this.addendum = builder.addendum;
+  }
+
+  /**
+   * @return the audit log context object from the enforcement response
+   */
+  public AuditLogContext getAuditLogContext() {
+    return auditLogContext;
+  }
+
+  /**
+   * @return if the enforcement/ authorization was allowed
+   */
+  public AuthorizationStatus isAuthorized() {
+    return authorized;
+  }
+
+  public UnauthorizedException createUnAuthorizedException() {
+    return new UnauthorizedException(principal, missingPermissions, entity, ex, requiresAllPermissions,
```

Comment:
Move it to the utility class. This are implementation details and should not be in the Spi